Gemfile

```diff
@@ -1,12 +1,7 @@
 source "https://rubygems.org"
 
-group :development do
-  gem 'bundler', '>= 1.0'
-  gem 'rake', '10.5.0'
-  gem 'minitest', '~> 5.8', '>= 5.8.4'
-  gem 'pry', '>= 0.10.1'
-  gem 'travis', '>= 1.8.2'
-  gem 'jeweler', '>=2.0.1'
-  gem 'codeclimate-test-reporter'
-  gem 'simplecov'
-end
+gemspec
+
+gem "minitest"
```

Comment:
I think we can specify a minimum version for these gems, since they are dev-dependencies and not gem dependencies.

`gem 'minitest', '~> 5.8'` (or similar)

Reply:
The `Gemfile.lock` file is now versioned within git, so all developers will install, when they run the `bundler install` command, the same version of the libraries that the project requires for its development.

https://bundler.io/v2.5/man/bundle-install.1.html#THE-GEMFILE-LOCK

On the other hand, the dependencies of the gem when it is installed in other projects are marked by what is indicated in the gemspec, although this gem does not have any runtime dependencies.

Reply:
Thanks, yes I know how all this works.

However, I think it's useful that the `Gemfile` signals intent. For example `gem 'minitest', '~> 5.8' signals that we're not expecting minor updates to cause any trouble, so those are alright to update automatically, or without much thought, but major versions need to be more explicit.

https://stackoverflow.com/questions/9265213/should-i-specify-exact-versions-in-my-gemfile

https://www.reddit.com/r/rails/comments/sujdqk/best_practice_should_i_specify_versions_in_gem/

Can you put these back?

```ruby

gem 'minitest', '~> 5.8'
gem 'rake', '~> 13.1'
gem 'simplecov', '~> 0.22', require: false
```

Reply:
Yes, of course!